### ISSUE TITLE
Provide way to avoid `bettermc::mclapply` in `get_predictions`

### ISSUE THREAD

**Issue by brookslogan** [cmu-delphi/covidcast#599]
Currently, `get_predictions` always uses `bettermc::mclapply`, even with just 1 core selected or with `parallel_execution = FALSE` (this just sets the number of cores to 1).  This can complicate debugging in the case of metaforecasters (ensembles, calibrated forecasters, forecasters with sensorized covariates, etc.) that run `get_predictions` within their bodies, because, e.g., while running `debugger` on a debug dump and inspecting objects from stack frames, another debug dump can be generated and overwrite the first one, and not all frames from the traceback are selectable in the debugger.

We could
- add a parameter to just use `lapply` regardless of all other arguments, and make sure not to pass `lapply` any of the `bettermc::mclapply` extra arguments in this case, or
- see if any of the arguments rely on a feature of `bettermc::mclapply` (this isn't just having more than one core; there is also, e.g., retrying, and maybe other stuff), and if not, dispatch to `lapply` as above, making sure not to pass extra args)

**Comment by brookslogan** [comment]
Workaround:
```r
invisible(bettermc::mclapply) # make sure bettermc is loaded (but not necessarily attached)
if (!exists("mclapply_backup", asNamespace("bettermc"))) {
  bmc_mclapply_backup = bettermc::mclapply
  bmc_mclapply_replacement = rlang::new_function(
    rlang::fn_fmls(bettermc::mclapply),
    quote(lapply(X, FUN, ...))
  )
  assignInNamespace(ns="bettermc", "mclapply", bmc_mclapply_replacement)
}
```
The function construction part might be useful in a real solution, but messing with another package's namespace is messy and forbidden by CRAN.

**Comment by dshemetov** [comment]
Fixed with #614
